controllers/datasource_controller.go

```diff
@@ -280,9 +285,14 @@ func (r *GrafanaDatasourceReconciler) onDatasourceCreated(ctx context.Context, g
 		return err
 	}
 
+	variables, err := r.CollectVariablesFromSecrets(ctx, cr)
+	if err != nil {
```

Comment:
Could it be wroth writing some custom error or even update the status of the datasource that the controller was unable to get the secret or something like that?

Reply:
Currently you would see the error message only in the operator logs. But I agree that adding it to the status would be a nicer UX.